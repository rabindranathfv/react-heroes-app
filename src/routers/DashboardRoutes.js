import { Routes, Route } from 'react-router-dom';

import { Navbar } from './../Components/ui/Navbar';

import { DcScreen } from './../Components/dc/DcScreen';
import { MarvelScreen } from './../Components/marvel/MarvelScreen';
import { SearchScreen } from './../Components/search/SearchScreen';
import { HeroScreen } from './../Components/hero/HeroScreen';


export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">
                <Routes>
                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DcScreen />} />

                    <Route path="search" element={<SearchScreen />} />
                    <Route path="hero/:heroeId" element={<HeroScreen />} />

                    <Route path="/" element={<MarvelScreen />} />

                </Routes>
            </div>
        </>
    )
}
