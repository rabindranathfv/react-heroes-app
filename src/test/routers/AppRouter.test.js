import { AppRouter } from '../../routers/AppRouter';
import { mount } from 'enzyme';
import { AuthContext } from '../../auth/authContext';

const contextVal = {
    user: {
        logged: false
    }
}
describe('AppRouter:::', () => {

    let wrapper = mount(
    <AuthContext.Provider value={contextVal} > 
        <AppRouter />
    </AuthContext.Provider>
    );

    beforeEach(() => {
        wrapper = mount(
            <AuthContext.Provider value={contextVal} > 
                <AppRouter />
            </AuthContext.Provider>
            );
    });

    it('Should show login route if use is not authenticated', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Login');
    });
});