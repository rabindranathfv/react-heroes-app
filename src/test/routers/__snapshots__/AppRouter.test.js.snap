// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter::: Should show login route if use is not authenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
